src/Raven.Server/ServerWide/Commands/ConnectionStrings/PutConnectionStringCommand.cs

```diff
@@ -146,4 +149,91 @@ public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
+            {
+                ConnectionString.Identifier ??= ConnectionString.GenerateIdentifier();
+            }
+            catch (Exception e)
+            {
+                throw new RachisApplyException("Failed to generate AI connection string identifier", e);
+            }
+
+            InClusterValidation(record);
+
+            record.AiConnectionStrings[ConnectionString.Name] = ConnectionString;
+        }
+
+        private void InClusterValidation(DatabaseRecord databaseRecord)
+        {
+            try
+            {
+                if (databaseRecord == null)
+                    throw new RachisApplyException("Failed to get database record, but it is required for further validation");
+
+                if (string.IsNullOrWhiteSpace(ConnectionString.Identifier))
+                    throw new RachisApplyException("Connection string identifier must be set, but it is not");
+
+                if (ConnectionString.ValidateIdentifier(out var errors) == false)
+                    throw new RachisApplyException( $"Invalid identifier format. Validation errors:{Environment.NewLine} - {string.Join($"{Environment.NewLine} - ", errors)}");
+
+                var isUpdate = databaseRecord.AiConnectionStrings.TryGetValue(ConnectionString.Name, out var oldAiConnectionString) && oldAiConnectionString != null;
+                var identifierConflicts = databaseRecord?.AiConnectionStrings
+                    .Where(x => x.Value != null && x.Value.Identifier == ConnectionString.Identifier && x.Key != ConnectionString.Name)
+                    .ToArray();
+
+                if (identifierConflicts.Length > 0)
+                    throw new RachisApplyException(
+                        $"Can't {(isUpdate ? "update" : "create")} connection string: '{ConnectionString.Name}'. " +
+                        $"The identifier '{ConnectionString.Identifier}' is already used by " +
+                        $"connection string{(identifierConflicts.Length > 1 ? "s" : "")} " +
+                        $"'{string.Join("', '", identifierConflicts.Select(x => x.Key))}'");
+
+                var etlsUsingConnection = databaseRecord.EmbeddingsGenerations.Where(x => x.ConnectionStringName == ConnectionString.Name).ToArray();
+                var isConnectionStringInUse = etlsUsingConnection.Length > 0;
+
+                if (isUpdate == false || isConnectionStringInUse == false)
+                    return;
+
+                var differences = ConnectionString.Compare(oldAiConnectionString);
+                if (differences.HasFlag(AiSettingsCompareDifferences.RequiresEmbeddingsRegeneration) == false)
+                    return;
+
+                var etlNames = string.Join("', '", etlsUsingConnection.Select(x => x.Name));
+                throw new RachisApplyException(
+                    $"Cannot update connection string '{ConnectionString.Name}' because it contains changes that would affect the structure or creation process of embeddings. " +
```

Comment:
Include the `differences` in the message

Reply:
done